### PR TITLE
Improvements to instant search

### PR DESCRIPTION
**Pull request by msasikanth**
https://app.clubhouse.io/simpledotorg/story/2891/improvements-to-instant-search

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/instantsearch/InstantSearchScreen.kt [hunk 1]
```diff
@@ -287,7 +287,13 @@ class InstantSearchScreen :
         .textChanges()
         .skipInitialValue()
         .debounce(500, TimeUnit.MILLISECONDS)
-        .map { SearchQueryChanged(it.toString()) }
+        .map { searchQuery ->
+          val trimmedSearchQuery = searchQuery
+              .trim()
+              .toString()
```
Comment (janhavisinghh) [inline app/src/main/java/org/simple/clinic/instantsearch/InstantSearchScreen.kt:293]: Do we need to convert this to toString? Isn't searchQuery a string already?

Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/instantsearch/InstantSearchScreen.kt:293]: It's a charsequence, so we need to convert it to string in the end.

**Review by janhavisinghh — APPROVED** [review]

**Pull request merged**